app/src/main/java/com/example/drawingactivity/DrawableAreaView.kt

```diff
@@ -0,0 +1,81 @@
+package com.example.drawingactivity
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.graphics.*
+import android.util.AttributeSet
+import android.view.MotionEvent
+import android.view.View
+
+@SuppressLint("ClickableViewAccessibility")
+class DrawableAreaView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0
+) : View(context, attrs, defStyle) {
+
+    var onBitmapDrawn: ((Bitmap) -> Unit)? = null
+    var bitmap: Bitmap? = null
+    private var shape: Rect? = null
+
+    var pathColor = Color.BLACK
+    private var pathPaint = Paint().apply {
+        color = pathColor
+        style = Paint.Style.STROKE
+        strokeWidth = 20f
+        strokeJoin = Paint.Join.ROUND
+        strokeCap = Paint.Cap.ROUND
+    }
+
+    private var bitmapPaint = Paint()
+    lateinit var path: Path
+
+    init {
+        setOnTouchListener { _, mEvent ->
+            when (mEvent.action) {
+                MotionEvent.ACTION_DOWN -> {
+                    pathPaint.color = pathColor
+                    path = Path()
+                    path.moveTo(mEvent.x, mEvent.y)
+                    invalidate()
+                }
+                MotionEvent.ACTION_MOVE -> {
+                    path.lineTo(mEvent.x, mEvent.y)
+                    invalidate()
+                }
+                MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> {
+                    path.lineTo(mEvent.x, mEvent.y)
+                    bitmap?.let {
+                        onBitmapDrawn?.invoke(it)
```

Comment:
What is this for? You haven't _drawn_ anything yet here? Kdoc will help me understand

Reply:
Removed because it didn't actually make any sense to invoke there over invoking at the end of onDraw
I also updated the Kdoc to hopefully make this less of a mess to follow